spec/models/dialog_group_spec.rb

```diff
@@ -21,4 +21,45 @@
       expect(dialog_group.dialog_fields).to be_empty
     end
   end
+
+  context '#update_dialog_tabs' do
+    before(:each) do
+      @dialog_group = FactoryGirl.create(:dialog_group)
+      @dialog_fields = FactoryGirl.create_list(:dialog_field, 2)
+      @dialog_group.dialog_fields << @dialog_fields
+    end
+
+    it 'deletes a dialog tab' do
+      fields = [
+        { 'id' => @dialog_fields.first.id }
+      ]
+
+      expect do
+        @dialog_group.update_dialog_fields(fields)
+      end.to change(@dialog_group.reload.dialog_fields, :count).by(-1)
+    end
+
+    it 'adds a dialog tab' do
+      fields = [
+        { 'id' => @dialog_fields.first.id },
+        { 'id' => @dialog_fields.last.id },
+        { 'name' => 'new tab', 'label' => 'new label' }
+      ]
+
+      expect do
+        @dialog_group.update_dialog_fields(fields)
+      end.to change(@dialog_group.reload.dialog_fields, :count).by(1)
+    end
+
+    it 'updates a dialog tab' do
+      fields = [
+        { 'id' => @dialog_fields.first.id, 'name' => 'updated name'},
+        { 'id' => @dialog_fields.first.id }
+      ]
+
+      expect do
+        @dialog_group.update_dialog_fields(fields)
+      end.to change(@dialog_fields.first.reload, :name)
+    end
+  end
```

Comment:
We probably want two contexts here, one for when the fields have the 'id' key, and one for without, because right now the branch for without isn't being tested.